modules/era/lua/wish_upon_a_star.lua

```diff
@@ -0,0 +1,34 @@
+-----------------------------------
```

Comment:
Move this whole thing to a `quest_workarounds` module, where `xi.module.modifyInteractionEntry('scripts/quests/bastok/Wish_Upon_a_Star', function(quest)` is just a single part of it, with the comments above it.

```
m:addOverride('xi.server.onServerStart', function()
    super()

-- Module to make Wish Upon a Star completable.
-- TODO include NONE in weather for bastok markets.
-- https://github.com/LandSandBoat/server/pull/5831 comments for fix action
-- Currently, Bastok does not have a "NONE" weather
-- Therefore, the quest cannot be completed.
-- Using this module, will allow the quest to be completed
-- Using "Sunny" weather instead.
    xi.module.modifyInteractionEntry('scripts/quests/bastok/Wish_Upon_a_Star', function(quest)
        quest.sections[2][xi.zone.BASTOK_MARKETS]['Enu'].onTrade = function(player, npc, trade)
        local isNight = VanadielTOTD() == xi.time.NIGHT or VanadielTOTD() == xi.time.MIDNIGHT
            if npcUtil.tradeHasExactly(trade, xi.item.FALLEN_STAR) then
                if
                    player:getWeather() == xi.weather.SUNSHINE and
                    isNight
                then
                    return quest:progressEvent(334)
                else
                    return quest:event(337)
                end
            end
        end
    end)
end)
```

Something like this, but formatted correctly, etc.

We don't want an explosion of modules for no reason